mmc-google-analytics/src/test/unit/java/info/novatec/metricscollector/google/DataProvider.java

```diff
@@ -1,17 +1,85 @@
 package info.novatec.metricscollector.google;
 
-/**
- * The purpose of this class is to provide test data
- */
+import java.util.ArrayList;
+import java.util.List;
+
+
 public class DataProvider {
-    //Metrics data
-    static final String UNIQUE_PAGE_VIEWS = "ga:uniquePageviews";
-    static final String BOUNCE_RATE = "ga:bounceRate";
-    static final String AVG_SESSION_DURATION = "ga:avgSessionDuration";
-    static final String TEST_METRIC = "ga:testmetric";
-
-    //Dimensions data
-    static final String HOST_NAME = "ga:hostname";
-    static final String PAGE_PATH = "ga:pagePath";
-    static final String TEST_DIMENSION = "ga:testdimension";
+
+    static final String GA_PAGE_PATH = "ga:pagePath";
+    static final String PAGE_PATH = "pagePath";
+    public static final String PAGE_PATH_VALUE = "/?testPagePath";
+
+    static final String GA_METRIC = "ga:metric";
+    static final String METRIC = "metric";
+    static final Double METRIC_VALUE = 1.12;
+
+    static final String GA_DIMENSION = "ga:dimension";
+    static final String DIMENSION_VALUE = "aDimension";
+
+    public static final String MEASUREMENT_NAME_VALUE = "testMeasurement";
+
+    public static final Long TIME_VALUE = 123l;
+
+    public static Metrics fillMetrics(Metrics metrics) {
+        metrics.setPagePath(PAGE_PATH_VALUE);
+        metrics.getDimensions().put(GA_PAGE_PATH, PAGE_PATH_VALUE);
+        metrics.getMetrics().put(GA_METRIC, METRIC_VALUE);
+        return metrics;
+    }
+
+    static List<String> createGoogleAnalyticsMetrics(int numberOfMetrics) {
+        List<String> metrics = new ArrayList<>();
+        for (int i = 1; i <= numberOfMetrics; i++) {
+            metrics.add(GA_METRIC + i);
+        }
+        return metrics;
+    }
+
+    static List<String> createGoogleAnalyticsDimensions(int numberOfDimensions) {
+        List<String> dimensions = new ArrayList<>();
+        for (int i = 1; i <= numberOfDimensions; i++) {
+            dimensions.add(GA_DIMENSION + i);
+        }
+        return dimensions;
+    }
+
+    public static Metrics createMetrics() {
+        return fillMetrics(createEmptyMetrics());
+    }
+
+    public static Metrics createEmptyMetrics() {
+        return new Metrics();
+    }
+
+    public static List<Metrics> createMetricsForPages(int numberOfPages) {
+        List<Metrics> metricsForPages = new ArrayList<>();
+        for (int numberOfCreatedMetrics = 0; numberOfCreatedMetrics < numberOfPages; numberOfCreatedMetrics++) {
+            metricsForPages.add(createMetrics());
+        }
+        return metricsForPages;
+    }
+
+    public static String createInfluxPointAsString() {
+        return "Point [name=" + MEASUREMENT_NAME_VALUE + ", time=" + TIME_VALUE + ", " + "tags={" + PAGE_PATH + "="
+            + PAGE_PATH_VALUE + "}, " + "precision=MILLISECONDS, " + "fields={" + METRIC + "=" + METRIC_VALUE + "}]";
+    }
+
+    public static List<String> getRequiredMetrics(){
```

Comment:
Since this (and also the method for getRequiredDimension()) is used as a test expectation criteria, the suggestion is the following:
1) When used in a single test class - it can be moved there.
2) When used in more test classes - a shared expected result data provider can be created.
Naming - no method is needed, only an ArrayList with name: expectedRequiredMetrics.

Reply:
As we discussed it. I moved it to the tests.